kymatio/backend/torch_backend.py

```diff
@@ -182,7 +183,7 @@ def cdgmm(A, B, inplace=False):
     if A.shape[-len(B.shape):-1] != B.shape[:-1]:
         raise RuntimeError('The filters are not compatible for multiplication.')
 
-    if A.dtype is not B.dtype:
+    if A.dtype is not B.dtype or type(A) is not type(B):
```

Comment:
I had to add this so [this](https://github.com/kymatio/kymatio/pull/551/files#diff-3ee77cd2d2c53e9935b29a827d4f7a0fR123) test would pass. Should we keep one or the other, or both?

Reply:
How was it passing before? We should be checking the `dtype`, not the type (since that's influenced by other things).

Reply:
The 3d test has a different type check, and I don't believe it has a check for dtype, just type.

Reply:
I suppose I should change the test/code so that we just make sure its the dtype thats checked.

Reply:
:+1: